### PR TITLE
make the Schwartz @safe

### PR DESCRIPTION
**Pull request by WalterBright**
At least make the schwartzSort() safe.

Partial fix for https://issues.dlang.org/show_bug.cgi?id=13054

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by tsbockman** [comment]
LGTM.

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -2058,25 +2058,35 @@ schwartzSort(alias transform, alias less = "a < b",
...
     if (isRandomAccessRange!R && hasLength!R)
 {
-    import core.stdc.stdlib : malloc, free;
     import std.conv : emplace;
     import std.string : representation;
     import std.range : zip, SortedRange;
 
     alias T = typeof(unaryFun!transform(r.front));
-    auto xform1 = (cast(T*) malloc(r.length * T.sizeof))[0 .. r.length];
+    static trustedMalloc(size_t len) @trusted
```
Comment (burner) [inline std/algorithm/sorting.d]: return type?

Comment (DmitryOlshansky) [inline std/algorithm/sorting.d]: Auto-deduced

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

File: std/algorithm/sorting.d (hunk 1 above)
Comment (burner) [inline std/algorithm/sorting.d]: where is the documentation for this?

**Pull request merged**